### PR TITLE
Целикова Арина. Задача 1. Вариант 2. Вычисление среднего значения элементов вектора.

### PR DESCRIPTION
**Pull request by ArinaTs**
Описание последовательной задачи:
Последовательно проходим по всем элементам вектора, считая их сумму. Затем делим на число элементов в векторе.

Описание MPI задачи:
Заданный вектор делим на подвектора, каждый из них передаем различным потокам. Значения, полученные из потоков складываем между собой внутри главного потока. Затем делим на число элементов вектора.

### CODE REVIEW

File: tasks/mpi/tselikova_a_average_of_vector_elements/src/ops_mpi.cpp [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  broadcast(world, total_elements, 0);
+
+  if (world.rank() == 0) {
+    input_ = std::vector<int>(taskData->inputs_count[0]);
+    auto* tmp_ptr = reinterpret_cast<int*>(taskData->inputs[0]);
+    for (unsigned i = 0; i < taskData->inputs_count[0]; i++) {
+      input_[i] = tmp_ptr[i];
+    }
+    for (int proc = 1; proc < world.size(); proc++) {
+      world.send(proc, 0, input_.data() + proc * delta, delta);
```
Comment (hasanyanovk) [inline tasks/mpi/tselikova_a_average_of_vector_elements/src/ops_mpi.cpp:63]: invalid distribution of elements to processes

File: tasks/seq/tselikova_a_average_of_vector_elements/func_tests/main.cpp [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+  taskDataSeq->outputs_count.emplace_back(out.size());
+
+  // Create Task
+  tselikova_a_average_of_vector_elements::TestTaskSequential testTaskSequential(taskDataSeq);
+  ASSERT_EQ(testTaskSequential.validation(), true);
+  testTaskSequential.pre_processing();
+  testTaskSequential.run();
+  testTaskSequential.post_processing();
+  ASSERT_EQ(-6, out[0]);
+}
```
Comment (VasenkovAA) [inline tasks/seq/tselikova_a_average_of_vector_elements/func_tests/main.cpp:172]: Add a test for a vector whose average element value is zero.

**Review by VasenkovAA — APPROVED** [review]

File: tasks/mpi/tselikova_a_average_of_vector_elements/include/ops_mpi.hpp [hunk 3]
```diff
@@ -0,0 +1,47 @@
+// Copyright 2023 Nesterov Alexander
+#pragma once
+
+#include <gtest/gtest.h>
+
+#include <boost/mpi/collectives.hpp>
+#include <boost/mpi/communicator.hpp>
+#include <memory>
+#include <numeric>
+#include <string>
```
Comment (Irina2004-tech) [inline tasks/mpi/tselikova_a_average_of_vector_elements/include/ops_mpi.hpp:10]: remove the library if you are not using it

File: tasks/seq/tselikova_a_average_of_vector_elements/include/ops_seq.hpp [hunk 4]
```diff
@@ -0,0 +1,24 @@
+// Copyright 2024 Tselikova Arina
+#pragma once
+
+#include <string>
```
Comment (Irina2004-tech) [inline tasks/seq/tselikova_a_average_of_vector_elements/include/ops_seq.hpp:4]: remove the library if you are not using it

**Review by Irina2004-tech — APPROVED** [review]

File: tasks/seq/tselikova_a_average_of_vector_elements/func_tests/main.cpp (hunk 2 above)
Comment (ArinaTs) [reply tasks/seq/tselikova_a_average_of_vector_elements/func_tests/main.cpp:193]: fixed

File: tasks/mpi/tselikova_a_average_of_vector_elements/include/ops_mpi.hpp (hunk 3 above)
Comment (ArinaTs) [reply tasks/mpi/tselikova_a_average_of_vector_elements/include/ops_mpi.hpp]: fixed

File: tasks/seq/tselikova_a_average_of_vector_elements/include/ops_seq.hpp (hunk 4 above)
Comment (ArinaTs) [reply tasks/seq/tselikova_a_average_of_vector_elements/include/ops_seq.hpp]: fixed

File: tasks/mpi/tselikova_a_average_of_vector_elements/src/ops_mpi.cpp [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+bool tselikova_a_average_of_vector_elements_mpi::TestMPITaskSequential::post_processing() {
+  internal_order_test();
+  std::cout << res << std::endl;
+  reinterpret_cast<float*>(taskData->outputs[0])[0] = res;
+  return true;
+}
+
+bool tselikova_a_average_of_vector_elements_mpi::TestMPITaskParallel::pre_processing() {
```
Comment (allnes) [inline tasks/mpi/tselikova_a_average_of_vector_elements/src/ops_mpi.cpp:45]: please use all mpi functions in run()

**Review by allnes — CHANGES_REQUESTED** [review]

File: tasks/mpi/tselikova_a_average_of_vector_elements/src/ops_mpi.cpp (hunk 5 above)
Comment (ArinaTs) [reply tasks/mpi/tselikova_a_average_of_vector_elements/src/ops_mpi.cpp:45]: fixed

**Review by aobolensk — APPROVED** [review]

**Review by allnes — APPROVED** [review]
